smack-core/src/main/java/org/jivesoftware/smack/AbstractXMPPConnection.java

```diff
@@ -1133,15 +1133,15 @@ public void run() {
                             }
                         }
                     });
-                    // The following returns makes it impossible for packet listeners and collectors to
-                    // filter for IQ request stanzas, i.e. IQs of type 'set' or 'get'. This is the
-                    // desired behavior.
-                    return;
                 }
                 break;
             default:
                 break;
             }
+            // The following returns makes it impossible for packet listeners and collectors to
```

Comment:
This makes some unit test fail. Most likely due interference with ThreadedDummyConnection.